sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/implementation/Utility.java

```diff
@@ -122,37 +120,6 @@ private static boolean isBmp(byte[] header) {
         return (header[0] == (byte) 0x42 && header[1] == (byte) 0x4D);
     }
 
-    /**
-     * Creates a Flux of ByteBuffer, with each ByteBuffer wrapping bytes read from the given
-     * InputStream.
-     *
-     * @param inputStream InputStream to back the Flux
-     *
-     * @return Flux of ByteBuffer backed by the InputStream
-     */
-    public static Flux<ByteBuffer> toFluxByteBuffer(InputStream inputStream) {
-        Pair pair = new Pair();
-        return Flux.just(true)
-            .repeat()
-            .map(ignore -> {
-                byte[] buffer = new byte[BYTE_BUFFER_CHUNK_SIZE];
-                try {
-                    int numBytes = inputStream.read(buffer);
-                    if (numBytes > 0) {
-                        return pair.buffer(ByteBuffer.wrap(buffer, 0, numBytes)).readBytes(numBytes);
-                    } else {
-                        return pair.buffer(null).readBytes(numBytes);
-                    }
-                } catch (IOException ioe) {
-                    throw LOGGER.logExceptionAsError(new RuntimeException(ioe));
-                }
-            })
-            .takeUntil(p -> p.readBytes() == -1)
-            .filter(p -> p.readBytes() > 0)
-            .map(Pair::buffer)
-            .cache();
```

Comment:
From the other discussion, let's turn this into a simple wrapper around the `FluxUtil` API until support for making the resulting `Flux<ByteBuffer>` replay-able.

```java
public static Flux<ByteBuffer> toFluxByteBuffer(InputStream inputStream) {
    return FluxUtil.toFluxByteBuffer(inputStream)
        .cache()
        .map(ByteBuffer::duplicate); // Or .map(ByteBuffer::asReadOnlyBuffer)
}
```

Reply:
Yeah, I think that should be better.